### DISCUSSION TITLE
Not sure why my Foundry Cast doesn't seem to be working

### DISCUSSION THREAD

**Discussion by Glow-in-the-dark** [Cyfrin/foundry-full-course-cu#544] [Q&A]
![Screenshot 2023-08-05 at 2 26 20 AM](https://github.com/Cyfrin/foundry-full-course-f23/assets/2751458/70847b7e-a82d-478e-9b14-7aad561da1ac)

Not sure why my simple "cast" call is not working. Any idea how to debug this ?

**Comment by theirrationalone** [comment]
execute command below and then try again.

```terminal
source /Users/username(your)/.bashrc
```

then... execute this 👇 command...

```terminal
foundryup
```
then... execute this 👇command...
```terminal
cast --version
```

after this 👆 command, Terminal will show you something like this...

```terminal
cast 0.2.0 (8c48932 2023-07-09T00:04:12.656091700Z)
```

and if any of command(s) failed then that means you haven't installed foundry yet. So try to install foundry first!

try to execute this 👇 command into terminal...

```terminal
curl -L https://foundry.paradigm.xyz | bash
```
and then execute that two first commands i mentioned just above!

**Reply by theirrationalone** [reply]
If everything exhausted and nothing worked, then Please try this last one...

👇execute it in terminal...
```terminal
export http_proxy=http://127.0.0.1:7890 https_proxy=http://127.0.0.1:7890
```

Now after this 👆, Try to execute your cast commands....!

**Reply by Glow-in-the-dark** [reply]
Thank brother!

I tried the above but it doesn't work, my foundry is definitely installed and running which is why i can run my deploy scripts/test and everything. 
![Screenshot 2023-08-05 at 12 38 21 PM](https://github.com/Cyfrin/foundry-full-course-f23/assets/2751458/7a12cf69-a699-4586-986f-0a7f8873bbad)
Managed to deploy to live testnet too:
![Screenshot 2023-08-05 at 12 39 04 PM](https://github.com/Cyfrin/foundry-full-course-f23/assets/2751458/066d1b69-e824-4988-92a9-32f0ae2305f9)
Just that the cast does not work.

I tried your above solutions, but the issues seems to still remain =(
example of running the above commands, and the results:
![Screenshot 2023-08-05 at 12 42 01 PM](https://github.com/Cyfrin/foundry-full-course-f23/assets/2751458/bd330974-f331-48ef-bd65-1fa35af0b156)


any idea ? Thanks so much BTW.

**Reply by shikhar229169** [reply]
Was cast working previously?
If so try making a new folder and try it there.

**Reply by Glow-in-the-dark** [reply]
for the other lessons was working just fine. except this lesson.

My new folder, u mean a totally new repo? or i can do it under my same git repo, just another new folder?
(created another folder, i did a `forge init` and then do `cast --version`, and it works.

Interested to understand the rational of why it's not working too tho. and also thanks @shikhar229169

**Reply by shikhar229169** [reply]
cast should work globally everywhere, whether you do forge init or not. Whether you are in any folder which is related or not related to foundry it should work.

**Reply by Glow-in-the-dark** [reply]
that's weird, my "cast" doesn't work in one of my folder, but yet it works on another
![Screenshot 2023-08-05 at 2 20 15 PM](https://github.com/Cyfrin/foundry-full-course-f23/assets/2751458/8022a45f-c0d0-431b-b939-fff8c20f8eb1)
😱

**Reply by theirrationalone** [reply]
> that's weird, my "cast" doesn't work in one of my folder, but yet it works on another
> ![Screenshot 2023-08-05 at 2 20 15 PM](https://github.com/Cyfrin/foundry-full-course-f23/assets/2751458/8022a45f-c0d0-431b-b939-fff8c20f8eb1)
> 😱 

That could mean cast isn't set for global environment!?

**Reply by theirrationalone** [reply]
> > that's weird, my "cast" doesn't work in one of my folder, but yet it works on another
> > ![Screenshot 2023-08-05 at 2 20 15 PM](https://github.com/Cyfrin/foundry-full-course-f23/assets/2751458/8022a45f-c0d0-431b-b939-fff8c20f8eb1)
> > 😱 
> 
> That could mean cast isn't set for global environment!? 

Try to put all projects into different folder instead of Github folder in desktop, foundry could be specific for keyword github/Github.

**Reply by Glow-in-the-dark** [reply]
@theirrationalone How do u make it global? and how can i check?

**Reply by theirrationalone** [reply]
> @theirrationalone How do u make it global? and how can i check?

When you install foundry, foundry done that for you... 

I'm just guessing that foundry might trying to get cast directly from github because of your that "Github" folder. 

So i would suggest you to do this.. 

[Follow this](https://www.google.com/amp/s/www.debugpoint.com/failed-connect-raw-githubusercontent-com-port-443/%3famp=1)  and then try again.
